### ISSUE TITLE
Edition number is shown in wrong location in merge interface when Work title is a right‐to‐left language

### ISSUE THREAD

**Issue by Freso** [internetarchive/openlibrary#10115]
### Problem

When merging authors, the merge interface shows a handful of works by each author in the merge, along with edition counts and earliest years of publication. If the Work title is written in a right‐to‐left language (e.g., Arabic) but the interface is a left‐to‐right language (e.g., English), the number of editions get included in the rtl part before the text switches back to ltr.

![e.g., “مدرسة العذاب 1 edition, 2009”](https://github.com/user-attachments/assets/b74ace2c-9516-4b7f-ac07-5f15be161012)

### Reproducing the bug

1. Go to https://openlibrary.org/authors/merge?key=OL5077340A&key=OL11487744A (or another author merge where at least one work has a title in a rtl language)

* Expected behavior: edition count and year, e.g. “1 edition, 2009”, being shown at one end (to the left or to the right, dependent on language direction) and the title, e.g. “مدرسة العذاب”, at the other.
* Actual behavior: The edition count gets shown mixed in with the rtl language (e.g., “مدرسة العذاب 1”) while the edition stands on it own with release year and no count (e.g., “edition, 2009”).

### Context

- Browser (Chrome, Safari, Firefox, etc): Firefox 133.0
- OS (Windows, Mac, etc): Linux (Arch, btw)
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

#### Ponderings

This can be fixed by explicitly marking the `<span>` that holds the edition count and year with a `dir="ltr"`. However, this should obviously only be done for ltr languages, so this needs to be dependent on what interface language is currently being used. (Not sure if we currently support any non‐ltr languages, but that might come at some point in the future, so the fix for this should not make assumptions based on currently supported locales.)

#### Requirements Checklist
* [ ]

#### Related files

*

#### Stakeholders

*
<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by DachiCharkviani** [comment]
Hello, I would like to work on this issue!

**Comment by mekarpeles** [comment]
@DachiCharkviani can you propose how you would go about addressing this one and @cdrini can offer you feedback on your approach? Thank you for your willingness to take this one on.

**Comment by suvanbanerjee** [comment]
@DachiCharkviani are you still working on this ?

@mekarpeles @cdrini I would like to give this a try

**Comment by Freso** [comment]
@suvanbanerjee The link still works, but you have to be an OL librarian to have permission to access the page.

Also, given no response from @DachiCharkviani and no PR either, I think if you do what @mek asked (i.e., <q>propose how you would go about addressing this one</q>), you should be good to take over the issue.

**Comment by suvanbanerjee** [comment]
@Freso the bug is caused in the below line
https://github.com/internetarchive/openlibrary/blob/570330004d4307be37d8b7e4b13641cee4e7acdc/openlibrary/templates/merge/authors.html#L93-L99

to fix this I will add a function to figure out if a language is RTL or not, that can be figured out through unicodedata of string specifically the bidirectional property like the below 
```
def is_rtl(text):
    for char in text:
        bidi = unicodedata.bidirectional(char)
        if bidi in ('R', 'AL'): #rtl or arabic check 
            return True
    return False
```

based on this i will pass is_rtl here https://github.com/internetarchive/openlibrary/blob/570330004d4307be37d8b7e4b13641cee4e7acdc/openlibrary/plugins/upstream/merge_authors.py#L280-L285

and add style to change text direction

other thing is I can modify the template only to find the rtl character instead of passing it from server

one doubts i have is what if its a mix string say urdu+number or arabic+english letters as I am not a native speaker of the languages i dont know how RTL languages work IRL, please guild me on that

**Comment by mekarpeles** [comment]
@cdrini, I think @suvanbanerjee may be on the right track?

@suvanbanerjee I think it may be worth submitting a PR based on your solution, thank you for putting in the effort.

I'm going to leave it to @cdrini to handle assignment as he is the lead of this issue.

**Comment by suvanbanerjee** [comment]
@cdrini can you guide me on how to handle mixed data ?

**Comment by cdrini** [comment]
Yeah, we have much room for improvement with handling of RTL languages! Did some research with ChatGPT : https://chatgpt.com/share/67e17c67-f004-8012-916f-6b8e20345d30

I think what we want here is: to add `$:cond(is_rtl(doc['title']), 'dir="rtl")`, like so:

```html
<li>
    <a href="$doc['key']"
        target="new"
        title="$_('Open in a new window')"
        $:cond(is_rtl(doc['title']), 'dir="rtl")
    >$doc['title']
    </a> .... etc
```

Where 

```py
@public
def is_rtl(text: str) -> bool:
    # 'R' = Right-to-Left, 'AL' = Arabic Letter
    return any(unicodedata.bidirectional(char) in ('R', 'AL') for char in text)
```

Put this in openlibrary/plugins/upstream/utils.py . The `@public` decorator means we'll be able to access this from the template, like in the above html snippet.

Let me know if you hit any issues!
